.github/workflows/node-build.yml

```
@@ -159,6 +159,135 @@ jobs:
       - name: Run tests
         run: pnpm --filter integration run-tests
 
+  performance-test:
+    runs-on: ubuntu-22.04
+    needs: prerequisite
+    timeout-minutes: 5
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        uses: ./.github/workflows/rafiki/env-setup
+
+      - name: Set up k6
+        uses: grafana/setup-k6-action@v1
+
+      - name: Setup hosts
+        run: |
+          echo "127.0.0.1 cloud-nine-wallet-test-backend" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 cloud-nine-wallet-test-auth" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 happy-life-bank-test-backend" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 happy-life-bank-test-auth" | sudo tee -a /etc/hosts
+
+      - name: Build dependencies
+        run: pnpm --filter performance build:deps
+
+      - name: Bundle
+        run: pnpm --filter performance bundle
+
+      - name: Start test environment
+        run: pnpm --filter performance testenv:compose up --wait --build -d
+
+      - name: Start Mock Account Entity Services
+        run: nohup pnpm --filter performance start-mases > mases.log 2>&1 &
+
+      - name: Run performance tests
+        id: perf_test
+        run: |
+          pnpm --filter performance run-tests:testenv -k --vus 4 --duration 1m | tee performance.log
```

What do we want the test config to be?

I opted for 4VUs and 1M after testing some different values. I think we want to:

1. not make the duration longer than necessary. Just dont want to extend the time it takes CI to finish if we can help it. So I prefered more VUs for a shorter time to some extent. There may still be some room to increase duration if we want. This is taking ~3m while the docker scans are taking 4-5m
2. reliably complete the tests without failures as a baseline. But approaching the point where we do see issues. I saw some errors at 5VUs so I backed off.

I _think_ the hardware is determined by `runs-on: ubuntu-22.04` and should always be the same across runs.

---

```suggestion
          pnpm --filter performance run-tests:testenv -k -q --vus 4 --duration 1m | tee performance.log
```
if you just want to log the final output (and not the intermediate progress log)